tests/unit/test_rest_streaming.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import random
+from typing import List
+from unittest.mock import patch
+
+import proto
+import pytest
+import requests
+
+from google.api_core import rest_streaming
+
+
+class Composer(proto.Message):
+    given_name = proto.Field(proto.STRING, number=1)
+    family_name = proto.Field(proto.STRING, number=2)
+
+
+class Song(proto.Message):
```

Comment:
Can we also add special fields like:
int64 (serialized as string, parsed as int)
[Int32Value](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/wrappers.proto#L88)
Duration
Timestamp

Full list of special JSON mappings according to proto spec can be found here: https://developers.google.com/protocol-buffers/docs/proto3#json

Reply:
protoplus doesn't have Duration and Timestamp, but I've added other fields.

Reply:
Timestamp and Duration are special. We describe the fields using the `*_pb2.py` proto message types, and then proto-plus handles them as specific special cases.

e.g.
```python
from google.protobuf import duration_pb2
from google.protobuf import timestamp_pb2

import proto

class Event(proto.Message):
    start_time = proto.Field(timestamp_pb2.Timestamp, number=1)
    duration = proto.Field(duration_pb2.Duration, number=2)

import datetime

e = Event(start_time=datetime.datetime(), duration=datetime.timedelta())
```

Reply:
Added.